packages/iov-ethereum/src/ethereumconnection.ts

```diff
@@ -119,7 +120,7 @@ export class EthereumConnection implements AtomicSwapConnection {
   }
 
   private readonly pollIntervalMs: number;
-  private readonly rpcClient: HttpJsonRpcClient;
+  private readonly rpcClient: HttpJsonRpcClient | WsJsonRpcClient;
```

Comment:
Can we have a shared interface for those two?

Reply:
Definitely, this is the main thing I want to refactor in the current approach.